cunumeric/module.py

```diff
@@ -1314,33 +1361,45 @@ def _concatenate(
     dtype=None,
     casting="same_kind",
     common_info=None,
+    slicing_only=False,
 ):
     if axis < 0:
         axis += len(common_info.shape)
     leading_dim = _builtin_sum(arr.shape[axis] for arr in inputs)
     out_shape = list(common_info.shape)
     out_shape[axis] = leading_dim
 
-    out_array = ndarray(
-        shape=out_shape, dtype=common_info.dtype, inputs=inputs
-    )
+    if slicing_only:
+        slices = []
+        out_array = (out_shape, slices)
```

Comment:
This is both a very bad naming (since `out_array` isn't an array!) and also untyped coding, as `_concatenate` returns either a pair of a shape and a list or an ndarray. I think you should extract out the common code and separate the function for `slicing_only=True` from that for `slicing_only=False`.